pkg/cmd/dev/ui.go

```diff
@@ -0,0 +1,192 @@
+
+Replaces 'make ui-watch'.`,
+		Args: cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, commandLine []string) error {
+			// Create a context that cancels when OS signals come in
+			ctx, stop := signal.NotifyContext(d.cli.Context(), os.Interrupt, os.Kill)
+			defer stop()
+
+			isOss, err := cmd.Flags().GetBool(ossFlag)
+			if err != nil {
+				return err
+			}
+
+			// Build prerequisites for db-console and cluster-ui
+			args := []string{
+				"build",
+				"//pkg/ui/workspaces/db-console/src/js:crdb-protobuf-client",
+			}
+			if !isOss {
+				args = append(args, "//pkg/ui/workspaces/db-console/ccl/src/js:crdb-protobuf-client-ccl")
+			}
+			logCommand("bazel", args...)
+			err = d.exec.CommandContextInheritingStdStreams(ctx, "bazel", args...)
+
+			if err != nil {
+				log.Fatalf("failed to build UI watch prerequisites: %v", err)
+				return err
+			}
+
+			// Extract remaining flags
+			portNumber, err := cmd.Flags().GetInt16(portFlag)
+			if err != nil {
+				log.Fatalf("unexpected error: %v", err)
+				return err
+			}
+			port := fmt.Sprint(portNumber)
+			dbTarget, err := cmd.Flags().GetString(dbTargetFlag)
+			if err != nil {
+				log.Fatalf("unexpected error: %v", err)
+				return err
+			}
+			_, err = url.Parse(dbTarget)
+			if err != nil {
+				log.Fatalf("invalid format for --%s argument: %v", dbTargetFlag, err)
+				return err
+			}
+			secure := mustGetFlagBool(cmd, secureFlag)
+
+			// `yarn` is required to be on a user's path, since it won't run via Bazel
+			err = d.ensureBinaryInPath("yarn")
+			if err != nil {
+				return err
+			}
+
+			dirs, err := getUIDirs(d)
+			if err != nil {
+				log.Fatalf("unable to find cluster-ui and db-console directories: %v", err)
+				return err
+			}
+
+			// Start the cluster-ui watch task
+			nbExec := d.exec.AsNonBlocking()
+			err = nbExec.CommandContextInheritingStdStreams(ctx, "yarn", "--silent", "--cwd", dirs.clusterUI, "build:watch")
+			if err != nil {
+				log.Fatalf("Unable to watch cluster-ui for changes: %v", err)
+				return err
+			}
+
+			var webpackDist string
+			if isOss {
+				webpackDist = "oss"
+			} else {
+				webpackDist = "ccl"
+			}
+
+			args = []string{
+				"--silent",
+				"--cwd",
+				dirs.dbConsole,
+				"webpack-dev-server",
+				"--config", "webpack.app.js",
+				"--mode", "development",
+				// Polyfill WEBPACK_SERVE for webpack v4; it's set in webpack v5 via `webpack serve`
+				"--env.WEBPACK_SERVE",
+				"--env.dist=" + webpackDist,
+				"--env.target=" + dbTarget,
+				"--port", port,
+			}
+			if secure {
+				args = append(args, "--https")
+			}
+
+			// Start the db-console web server + watcher
+			err = nbExec.CommandContextInheritingStdStreams(ctx, "yarn", args...)
+			if err != nil {
+				log.Fatalf("Unable to serve db-console: %v", err)
+				return err
+			}
+
+			// Wait for OS signals to cancel if we're not in test-mode
+			if d.exec.Recording == nil {
```

Comment:
Nit: you might use `isTesting` instead.

Reply:
D'oh, great point!  I'll switch that out ASAP.